src/components/ScenarioParameters/ScenarioParameters.spec.js

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+
+import React from 'react';
+import { screen, fireEvent } from '@testing-library/react';
+import { ScenarioParameters } from '.';
+import { customRender } from '../../../tests/utils/renderInThemeAndStoreProviders';
+import { AccordionTesting } from '../../../tests/MuiComponentsTesting/AccordionTesting';
+
+const mockOnToggleAccordion = jest.fn();
+
+jest.mock('./components', () => ({
+  __esModule: true,
+  ScenarioParametersTabsWrapper: () => {
+    return <div data-testid="scenario_parameters_tabs_wrapper">Wrapper</div>;
+  },
+  ScenarioActions: () => {
+    return <div data-testid="scenario-actions" />;
+  },
+}));
+
+const Accordion = new AccordionTesting({ dataCy: 'scenario-params-accordion' });
+const getScenarioActionsButtons = () => screen.getByTestId('scenario-actions');
+const getTabsWrapper = () => screen.getByTestId('scenario_parameters_tabs_wrapper');
+
+const PROPS_WITH_ACCORDION_COLLAPSED = { onToggleAccordion: mockOnToggleAccordion, isAccordionExpanded: false };
+const PROPS_WITH_ACCORDION_EXPANDED = { onToggleAccordion: mockOnToggleAccordion, isAccordionExpanded: true };
+
+describe('accordion is expanded or collapsed', () => {
+  beforeEach(() => {
+    mockOnToggleAccordion.mockClear();
+  });
+
+  it('checks accordion is collapsed and prop function is called on click', () => {
+    customRender(<ScenarioParameters {...PROPS_WITH_ACCORDION_COLLAPSED} />);
+    expect(Accordion.Container).toBeVisible();
+    expect(Accordion.ToggleButton).toBeVisible();
+    expect(getScenarioActionsButtons()).toBeVisible();
+    expect(getTabsWrapper()).not.toBeVisible();
+    fireEvent.click(Accordion.ToggleButton);
+    expect(mockOnToggleAccordion).toHaveBeenCalled();
```

Comment:
maybe check if accordion is collapse after click ?

Reply:
It was the idea but then @ardvi pointed out that checking the function isn't in the scope of the test because the component folds and unfolds via props and doesn't have any control on the function and its results, so it's better to check that the component accepts the props and calls the function on click.